src/Psy/TabCompletion/AutoCompleter.php

```diff
@@ -62,6 +62,10 @@ public function processCallback($input, $index, $info = array())
             $line = $input;
         }
 
+        if ($this->shouldInsertOnlyTab($line, $info['point'])) {
+            return array("\t");
```

Comment:
because of the magic space inserted after tab completion, this inserts five spaces for the first tab stop, then nine spaces for each additional tab stop. why don't we hijack this and return three spaces? then it's always four space indents :)

Reply:
It inserts a tab character, not spaces. I'm definitely not a fan of replacing it with space characters, as those are different (no matter what you think of tabs vs spaces) and substituting one for the other silently can never be what you want.

The tab + space behaviour is sort of vexing, but it's not something I can realistically solve in a way that doesn't break everything in environments that do work properly (for example ones that have the path from https://github.com/php/php-src/pull/2720 applied). 

What I *can* probably do is `return array("\t" . chr(8));`. `chr(8)` is the backspace character, which should work.

Reply:
I could add in a setting for this and apply it to every autocompletion, thinking about it. Until the patch I mentioned works that's a sort of viable workaround to get autocompletion as you'd expect from an IDE.

Reply:
Wait no, that won't work. The space is appended after the completion - so a backspace that is inserted will actually just remove the tab character entirely.

Reply:
i understand that tabs and spaces are different. but practically, one looks like you'd expect, and the other looks really bad. so ¯\_(ツ)_/¯